statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -3910,6 +3910,19 @@ def plot_diagnostics(self, variable=0, lags=10, fig=None, figsize=None):
         # Eliminate residuals associated with burned or diffuse likelihoods
         d = np.maximum(self.loglikelihood_burn, self.nobs_diffuse)
         resid = self.filter_results.standardized_forecasts_error[variable, d:]
+        try:
```

Comment:
Don't use a try and except.  Just check if resd.size < d and then raise or warn as appropriate.  We never use prints.  The function should raise ValueError if the data size is too small.